### PR TITLE
Merge scala & java content for typed.md docs (#23052)

### PR DESCRIPTION
**Pull request by gosubpl**
As usual, two commits:

1. Unification of content for both scala/typed.md and java/typed.md that allows to clearly see the diff
2. Replacement of java/typed.md with a symlink to scala/typed.md

### CODE REVIEW

**Comment by gosubpl** [comment]
Refs #23052

File: akka-docs/src/main/paradox/java/typed.md [hunk 1]
```diff
@@ -83,16 +83,16 @@ also typed as such. This is why we can access the `whom` and `replyTo`
 members without needing to use a pattern match.
 
 On the last line we see the `HelloWorld` Actor send a message to another
-Actor, which is done using the `!` operator (pronounced “tell”). Since the
-`replyTo` address is declared to be of type `ActorRef<Greeted>` the
+Actor, which is done using the @scala[`tell` method (represented by the `!` operator)]@java[`!` operator (pronounced “tell”)].
```
Comment (raboof) [inline akka-docs/src/main/paradox/java/typed.md]: Do we have a `!` operator on Java at all?

File: akka-docs/src/main/paradox/scala/typed.md [hunk 2]
```diff
@@ -103,13 +103,15 @@ Java
...
 After importing the Actor’s protocol definition we start an Actor system from
-the defined behavior.
+the defined `greeter` behavior.
 
 As Carl Hewitt said, one Actor is no Actor—it would be quite lonely with
 nobody to talk to. In this sense the example is a little cruel because we only
 give the `HelloWorld` Actor a fake person to talk to—the “ask” pattern
 (represented by the `?` operator) can be used to send a message such that the
-reply fulfills a Promise to which we get back the corresponding Future.
+reply fulfills a @scala[Promise to which we get back the corresponding Future]@java[`CompletionStage`].
```
Comment (raboof) [inline akka-docs/src/main/paradox/scala/typed.md]: Perhaps wrap `Promise` in backticks

**Review by raboof — APPROVED** [review]
2 small comments to check, then LGTM!

File: akka-docs/src/main/paradox/java/typed.md (hunk 1 above)
Comment (gosubpl) [inline akka-docs/src/main/paradox/java/typed.md]: Huh, not really, thanks for catching this.

File: akka-docs/src/main/paradox/scala/typed.md (hunk 2 above)
Comment (gosubpl) [inline akka-docs/src/main/paradox/scala/typed.md]: Yup, thanks.

**Comment by gosubpl** [comment]
Comments reflected. Thanks @raboof for quick turnaround & sharp eye!

**Comment by raboof** [comment]
Likewise! :) - thanks!

**Pull request merged**